src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -5603,7 +5603,7 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
     <value>Explicit application of 'System.Runtime.CompilerServices.NullableAttribute' is not allowed.</value>
   </data>
   <data name="ERR_NullableUnconstrainedTypeParameter" xml:space="preserve">
-    <value>A nullable type parameter must be known to be a value type or non-nullable reference type. Consider adding a 'class', 'struct', or type constraint.</value>
+    <value>A nullable type parameter must be known to be a value type or non-nullable reference type. Please use language version '{0}' or greater, or consider adding a 'class', 'struct', or type constraint.</value>
```

Comment:
>Please use language version '{0}'  [](start = 102, length = 34)

I think we should simply use regular language version message with specific feature name, etc. This way IDE will be able to suggest the change language version fix.

Reply:
This message has helpful suggestions for using `T?` in C#8 which the general language version message would not. We can update the IDE to recognize this `ErrorCode` if necessary.

---
In reply to: [456600909](https://github.com/dotnet/roslyn/pull/45993#discussion_r456600909) [](ancestors = 456600909)

Reply:
> We can update the IDE to recognize this ErrorCode if necessary.

Let's do that then. I think we already have a mechanism to communicate to IDE that it should offer an upgrade to specific language version. CC @jcouv

---
In reply to: [456613552](https://github.com/dotnet/roslyn/pull/45993#discussion_r456613552) [](ancestors = 456613552,456600909)